src/main/java/com/vinyl/service/ProductService.java

```diff
@@ -66,4 +68,49 @@ private ProductCart createOrUpdateProductCart(Product product, Cart cart, Intege
 
         return productCartRepository.save(productCart);
     }
+
+    public void addProductToStore(ProductManagementDTO productToAdd, String tokenHash) {
+        validateAdminToken(tokenHash);
+        validatorFactory.getProductManagementValidator().validate(productToAdd);
+
+        Product product = setAndGetProduct(productToAdd);
+        productRepository.save(product);
+    }
+
+    public void updateProduct(ProductManagementDTO updatedProduct, String tokenHash, Integer productId) {
+        validateAdminToken(tokenHash);
+        validatorFactory.getProductManagementValidator().validate(updatedProduct);
+        validateProductId(productId);
+
+        Product newProduct = setAndGetProduct(updatedProduct);
```

Comment:
rename to createProductFrom or something else, its more natural than setAndGet

Reply:
done